Lib/concurrent/futures/thread.py

```diff
@@ -158,7 +161,7 @@ def __init__(self, max_workers=None, thread_name_prefix='',
         self._initializer = initializer
         self._initargs = initargs
 
-    def submit(self, fn, /, *args, **kwargs):
+    def submit(self, fn, name=None, /, *args, **kwargs):
```

Comment:
This is a backwards-incompatible change. It would break existing uses of `submit`. (For example, in current Python code, `executor.submit(int, "101", base=2)` calls `int("101", base=2)` in the background, giving a result of `5`. With this change, that same `submit` call calls `int(base=2)` in the background, giving a `TypeError`.

Reply:
yes,  it's not a good practice, I wanted to create a new method `submit_with_name` in _base `Executor` which resolves works like this;

``` python
class Executor(object):
    """This is an abstract base class for concrete asynchronous executors."""

    def submit_with_name(self, fn, name=None, /, *args, **kwargs):
        """Submits a callable to be executed with the given arguments and also with a name.

        Schedules the callable to be executed as fn(*args, **kwargs) and returns
        a Future instance representing the execution of the callable.

        Returns:
            A Future representing the given call.
        """

        raise NotImplementedError()

    def submit(self, fn, /, *args, **kwargs):
        """Submits a callable with default name

        Returns:
            A Future representing the given call.
        """
        return self.submit_with_name(fn, None, *args, **kwargs)
```

Can I proceed with this?